src/auth/user.repository.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2020-present Open Networking Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { HttpException, HttpStatus} from '@nestjs/common';
+import { EntityRepository, getConnection, Repository, UpdateResult } from 'typeorm';
+import { User } from './user.entity';
+import { CreateUserDto } from './dto/create-user.dto';
+import { UpdateUserDto } from './dto/update-user.dto';
+import { Project } from '../project/project.entity';
+
+@EntityRepository(User)
+export class UserRepository extends Repository<User> {
+
+  async createTsUser(createUserDto: CreateUserDto): Promise<void> {
+
+    const { email, firstName, lastName, supervisorEmail, darpaAllocationPct, isSupervisor, projects } = createUserDto;
+
+    const newUser = new User();
+    newUser.email = email;
+    newUser.firstName = firstName;
+    newUser.lastName = lastName;
+    newUser.supervisorEmail = supervisorEmail;
+    newUser.darpaAllocationPct = darpaAllocationPct;
+    newUser.isSupervisor = isSupervisor;
```

Comment:
a person might be a supervisor on one and not on the other

There are lots of complicated scenarios to think of